sphinxcontrib/openapi.py

```diff
@@ -122,10 +122,30 @@ def _httpresource(endpoint, method, properties):
     yield ''
 
 
-def openapi2httpdomain(spec):
+def _clean_paths_option(value):
+    """ Clean paths string option
+
+    Split string on newlines to get list of paths to filter by
+
+    Returns an array of string paths to filter on
+    Raises ValueError if unable to parse the input value
+
+    """
+    paths = [path.strip() for path in value.splitlines()]
+    if len(paths) > 0:
+        return paths
+    else:
+        raise ValueError('Invalid argument to paths param: {}'.format(value))
+
+
+def openapi2httpdomain(spec, **options):
     generators = []
 
-    for endpoint in spec['paths']:
+    # If we don't pass a specific list of paths to render,
+    # default to all in spec
+    paths = options.get('paths', spec['paths'])
+
```

Comment:
Could you please add a check that `paths` is a subset of `spec['paths']`, and raise some exception with a user-friendly message if not. You see, we don't control what user passes to this option, so failing with `KeyError` on line `149` is a very real case.